sentry-ruby/lib/sentry/session.rb

```diff
@@ -17,9 +16,11 @@ def initialize
       @aggregation_key = Time.utc(@started.year, @started.month, @started.day, @started.hour, @started.min)
     end
 
-    # TODO-neel add :crashed after adding handled mechanism
-    def update_from_exception(_exception = nil)
+    def update_from_event(event = nil)
       @status = :errored
+      return unless event&.is_a?(ErrorEvent)
+
+      @status = :crashed if event.exception.values.any? { |e| e.mechanism&.handled == false }
```

Comment:
From #2280, it looks like there will always be a mechanism in the exception interface, so we won't need `&` here?